bftengine/include/bftengine/KeyExchangeManager.hpp

```diff
@@ -192,6 +196,7 @@ class KeyExchangeManager {
   std::shared_ptr<IInternalBFTClient> client_;
   std::vector<IKeyExchanger*> registryToExchange_;
   IMultiSigKeyGenerator* multiSigKeyHdlr_{nullptr};
+  IClientPublicKeyStore* clientPublicKeyStore_{nullptr};
```

Comment:
Doesn't it contradict your shared_ptr philosophy :)

Reply:
It does, but I followed your pattern here :-)

Reply:
but you convinced me to adopt yours :)